packages/@uppy/core/src/UploaderPlugin.js

```diff
@@ -0,0 +1,22 @@
+import BasePlugin from './BasePlugin.js'
+
+export default class UploaderPlugin extends BasePlugin {
```

Comment:
I'm not sure if more inheritance makes sense here. Why can't everything regarding remote uploads live in companion client? Or can't this be solved with composition instead of inheritance?

Reply:
Because it needs to access stuff from `this`: `this.connectToServerSocket` and `this.uppy` as well as `queueRequestSocketToken`, all of wihch live inside a subclass.

By "companion client" I assume you mean RequestClient. Maybe we could refactor it into there, but I think I tried to do that and it turns out to be a bigger task, because all code that it depends on should also be moved in there then. So maybe not in this PR?

Reply:
I'm just afraid "maybe not in this PR" turns into never doing it. We're keep running into issues of duplication between uploaders. If we plan to tackle that with this in mind we can leave it as is :)

Reply:
Hopefully we are going to do it, because all the uploaders share lots of code so all in likelihood we are going to eventually come back to that code.